pulp_smash/tests/rpm/api_v2/test_content_sources.py

```diff
@@ -0,0 +1,167 @@
+"""
+import os
+import unittest
+from io import StringIO
+from urllib.parse import urlsplit, urlunsplit
+
+from pulp_smash import api, cli, config, utils
+from pulp_smash.constants import (
+    CONTENT_SOURCES_PATH,
+    PULP_FIXTURES_BASE_URL,
+)
+from pulp_smash.tests.rpm.utils import check_issue_1282
+from pulp_smash.tests.rpm.utils import set_up_module as setUpModule  # noqa pylint:disable=unused-import
+
+_HEADERS = {'X-RHUI-ID', 'X-CSRF-TOKEN'}
+
+
+class ValidHeadersTestCase(unittest.TestCase):
+    """Define a content source with valid headers.
+
+    See: `Pulp Smash #643 <https://github.com/PulpQE/pulp-smash/issues/643>`_.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a content source with valid headers."""
+        cls.cfg = config.get_config()
+        cls.cs_kwargs = {
+            'source_id': utils.uuid4(),
+            'headers': ' '.join(
+                '{}={}'.format(header, utils.uuid4()) for header in _HEADERS
+            ),
+        }
+        cs_body = _gen_content_source_body(**cls.cs_kwargs)
+        cls.cs_path = _gen_content_source(cls.cfg, cs_body)
+
+    def test_list_content_sources(self):
+        """List Pulp's content sources.
+
+        Assert that:
+
+        1. The list includes an entry corresponding to the newly defined
+           content source configuration file.
+        2. This liste entry includes the headers defined in the configuration
+           file.
+        """
+        client = api.Client(self.cfg, api.json_handler)
+        sources = client.get('/pulp/api/v2/content/sources/')
+        source_id = self.cs_kwargs['source_id']
+        sources_by_id = {source['source_id']: source for source in sources}
+        self.assertIn(source_id, sources_by_id)
+
+        source_headers = self.cs_kwargs['headers']
+        self.assertEqual(source_headers, sources_by_id[source_id]['headers'])
+
+    def test_refresh_content_sources(self):
+        """Refresh Pulp's content sources.
+
+        Assert that the refresh completes successfully.
+        """
+        client = api.Client(self.cfg)
+        client.post('/pulp/api/v2/content/sources/action/refresh/')
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroy the created content source."""
+        sudo = () if utils.is_root(cls.cfg) else ('sudo',)
+        cli.Client(cls.cfg).run(sudo + ('rm', '-f', cls.cs_path))
+
+
+class InvalidHeadersTestCase(unittest.TestCase):
+    """Define a content source with invalid headers.
+
+    See: `Pulp Smash #643 <https://github.com/PulpQE/pulp-smash/issues/643>`_.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a content source with valid headers."""
+        cls.cfg = config.get_config()
+        if check_issue_1282(cls.cfg):
+            raise unittest.SkipTest('https://pulp.plan.io/issues/1282')
+        cls.cs_kwargs = {
+            'source_id': utils.uuid4(),
+            'headers': ' '.join(header + utils.uuid4() for header in _HEADERS),
```

Comment:
So, what makes these headers "invalid" is that they have the header and the utils.uuid4() just concatenated together instead of having a '=' between them?